source/_posts/yaml-pipelines.md

```diff
@@ -0,0 +1,63 @@
+---
+title: YAML pipelines in Azure DevOps
+date: 2020-03-29 18:00:00
+tags: 
+- Azure DevOps
+- YAML
+- pipeline
+---
+
+In this blog post, we are going to talk about YAML pipelines in Azure DevOps. Microsoft has been pushing YAML as the new default way of setting up build/release pipelines for some time now. When it was first set up, documentation was lacking and not all tasks that were available in the visual designer were available in YAML. Thus, it makes sense that the developers who looked into it at that moment were put off. However, much has changed since then and YAML pipelines have grown into a mature framework of setting up your build/release pipelines. Let's take a look.
+
+## What is YAML?
+> **YAML** is a human friendly data serialization standard for all programming languages.
+
+The YAML language is very clean and, like Python, used indentation to indicate nesting. Like JSON, it uses `[]` for lists and `{}` for maps. It is often used as a language to define settings or configuration files. For an overview of the YAML language, take a look at the [YAML cheat sheet](https://yaml.org/refcard.html).
+
+## Why would you use YAML pipelines?
+Moving to have *everything as code* is making is in an uptrend, for good reasons. Having your build/deployment pipeline as code, in version control, gives you some powerful benefits over having it defined in the visual designer. Sure, you lose the purely visual representation as you *are* working in a text file, but you gain the following:
+
++ your pipeline can be validated through code reviews by your colleagues in pull-requests or in some other way
++ if a change is introduced that *does* break the pipeline, restoring it is as easy as reverting the commit
++ should a pipeline have to be set up again, it can be done by selecting the YAML definition and you're done
++ you can have separate YAML pipeline definitions in your code, per project, per solution, multiple per project, you decide
++ you can set up different templates, like tasks, and combine them, like in task groups, and re-use them very easily
++ environments and (conditional) multi-stage deployments are also defined in the YAML pipeline definition
+
+In the end, you can get to a situation where you don't need to look at the pipeline itself anymore to deploy your code, even with pipeline changes:
+1. edit your code
+2. edit the yaml file
+3. push to the repo
+4. Azure Pipelines runs
+5. deployed to target
+
+## How are Azure pipelines set up?
+Azure pipelines contain the following key concepts:
++ *triggers* tell pipelines to run;
++ *stages* are used to organize jobs, can require approvals;
++ *environments* are the targets for deployment and can define their own deployment strategies and resources
++ *jobs* are used to organize steps, run on an agent or can run agentless
++ *agents* are installable software running a job,
++ *steps* are tasks or scripts, the smallest item of a pipeline
```

Comment:
zijn task niet de smallest item of a pipeline?
steps are part of a job organisation?